tools/github_webhook_processor.php

```diff
@@ -223,7 +235,91 @@ function handle_pr($payload) {
 	
 	$msg = '['.$payload['pull_request']['base']['repo']['full_name'].'] Pull Request '.$action.' by '.htmlSpecialChars($payload['sender']['login']).': <a href="'.$payload['pull_request']['html_url'].'">'.htmlSpecialChars('#'.$payload['pull_request']['number'].' '.$payload['pull_request']['user']['login'].' - '.$payload['pull_request']['title']).'</a>';
 	sendtoallservers('?announce='.urlencode($msg), $payload);
+}
+
+function create_comment($payload, $comment){
+	$scontext = array('http' => array(
+		'method'	=> 'POST',
+		'header'	=>
+			"Content-type: application/json\r\n".
+			'Authorization: token ' . $apiKey,
+		'ignore_errors' => true,
+		'user_agent' 	=> 'tgstation13.org-Github-Automation-Tools',
+		'content' => json_encode(array('body' => $comment))
+	));
+	echo file_get_contents($payload['pull_request']['comments_url'], false, stream_context_create($scontext));
+}
+
+function get_pr_labels_array($payload){
+	$url = $payload['pull_request']['issue_url'];
+
+	$scontext = array('http' => array(
+		'method'	=> 'GET',
+		'header'	=>
+			"Content-type: application/json\r\n".
+			'Authorization: token ' . $apiKey,
+		'ignore_errors' => true,
+		'user_agent' 	=> 'tgstation13.org-Github-Automation-Tools'
+	));
 
+	$issue = json_decode(file_get_contents($url, false, stream_context_create($scontext)), true);
+	$result = array();
+	if(isset($issue['labels']))
+		foreach($issue['labels'] as $l)
+			$result[] = $l['name'];
+	return $result;
+}
+
+function pr_balance_json_path(){
+	return $prBalanceJson != '' ? $prBalanceJson : 'pr_balances.json';
+}
+
+function pr_balances(){
+	$path = pr_balance_json_path();
+	if(file_exists($path))
+		return json_decode(file_get_contents($path), true);
+	else
+		return array();
+}
+
+//returns 1 if it's a fix/feature/etc, 0 if it's a neutral, -1 if it's a Feature/Tweak/Balance
+function get_pr_code_friendliness($payload){
+	$labels = get_pr_labels_array($payload);
+	//doing one of these increases your positive score
+	$improvement_labels = array('Fix', 'Refactor', 'Code Improvement', 'Priority: High', 'Priority: CRITICAL', 'Atmospherics', 'Grammar and Formatting', 'Logging', 'Performance');
+	//doing one of these prevents your negative score from increasing
+	$neutral_labels = array('Tools', 'Map Edit', 'SQL', 'Documentation', 'Repository', 'Revert/Removal', 'UI', 'Sprites', 'Sound');
```

Comment:
@MrStonedOne in regards to your comment below, $improvement_labels gives you positive score, $neutral_labels give you zero score, all other labels give you -1 score